src/test/java/org/trebol/operation/controllers/ControllersCorsTest.java

```diff
@@ -0,0 +1,235 @@
+		List<Arguments> args = new ArrayList<>();
+		
+		for (ApiPath apiPath : apiPaths) {
+			for (String method : apiPath.methods) {
+				for (String origin : allowedOrigins) {
+					args.add(arguments(apiPath.getPath(), method, origin));
+				}
+			}
+		}
+		
+		return args.stream();
+	}
+	
+	// All paths with all methods and all not allowed origins
+	private Stream<Arguments> provideArgumentsWithNotAllowedOrigins() {
+		List<Arguments> args = new ArrayList<>();
+		
+		for (ApiPath apiPath : apiPaths) {
+			for (String method : apiPath.methods) {
+				for (String origin : notAllowedOrigins) {
+					args.add(arguments(apiPath.getPath(), method, origin));
+				}
+			}
+		}
+		
+		return args.stream();
+	}
+	
+	// All paths with all methods and all allowed headers
+	private Stream<Arguments> provideArgumentsWithAllowedHeaders() {
+		List<Arguments> args = new ArrayList<>();
+		
+		for (ApiPath apiPath : apiPaths) {
+			for (String method : apiPath.methods) {
+				for (String header : allowedHeaders) {
+					args.add(arguments(apiPath.getPath(), method, header));
+				}
+			}
+		}
+		
+		return args.stream();
+	}
+	
+	// Some Paths with Invalid methods
+	private Stream<Arguments> provideArgumentsWithInvalidMethods() {
+		List<Arguments> args = new ArrayList<>();
+		
+		for (ApiPath apiPath : apiPathsWithInvalidMethods) {
+			for (String method : apiPath.methods) {				
+				args.add(arguments(apiPath.getPath(), method));				
+			}
+		}
+		
+		return args.stream();
+	}
+	
+	// Tests
+	
+	@ParameterizedTest
+	@MethodSource("provideArgumentsWithAllowedOrigins")
+	void whenAllowedOrigins_thenStatusOk(String path, String method, String origin) throws Exception {
+		mvc.perform(options(path)
+				.header("Access-Control-Request-Method", method)
+				.header("Origin", origin))			
+			.andExpect(status().isOk());
+	}
+	
+	@ParameterizedTest
+	@MethodSource("provideArgumentsWithNotAllowedOrigins")
+	void whenNotAllowedOrigins_thenStatus403(String path, String method, String origin) throws Exception {
+		mvc.perform(options(path)
+				.header("Access-Control-Request-Method", method)
+				.header("Origin", origin))			
+			.andExpect(status().is(403));
+	}
+	
+	@ParameterizedTest
+	@MethodSource("provideArgumentsWithAllowedHeaders")
+	void whenAllowedHeaders_thenStatusOk(String path, String method, String header) throws Exception {
+		String origin = allowedOrigins.get(0);
+
+		mvc.perform(options(path)
+				.header("Access-Control-Request-Headers", header)
+				.header("Access-Control-Request-Method", method)
+				.header("Origin", origin))
+			.andExpect(status().isOk());
+	}
+	
+	@ParameterizedTest
+	@MethodSource("provideArgumentsWithInvalidMethods")
+	void whenInvalidMethods_thenStatus403(String path, String method) throws Exception {
+		String origin = allowedOrigins.get(0);
+		
+		mvc.perform(options(path)
+				.header("Access-Control-Request-Method", method)
+				.header("Origin", origin))			
+			.andExpect(status().is(403));
+	}
+	
+	// Helper Class
+	
```

Comment:
Please turn this into a class JavaDoc comment
```suggestion
	/**
	 * Helper class that holds execution data for testing a single RESTful API call
	 */
```